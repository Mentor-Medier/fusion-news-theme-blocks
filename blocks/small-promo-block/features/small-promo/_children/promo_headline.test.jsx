import React from 'react';
import { mount } from 'enzyme';
import PromoHeadline from './promo_headline';

const { default: mockData } = require('../mock-data');

jest.mock('fusion:themes', () => (jest.fn(() => ({}))));

jest.mock('fusion:content', () => ({
  useContent: jest.fn(() => (mockData)),
  useEditableContent: jest.fn(() => ({ editableContent: () => ({ contentEditable: 'true' }) })),
}));

jest.mock('fusion:context', () => ({
  useFusionContext: jest.fn(() => ({})),
}));

describe('promo headline', () => {
  beforeEach(() => {
    jest.mock('fusion:context', () => ({
      useFusionContext: jest.fn(() => ({
        arcSite: 'the-sun',
        id: 'testId',
      })),
    }));
  });

  afterEach(() => {
    jest.resetModules();
  });

  it('return null while no content available in props', () => {
    const mockCustomFields = {
      imagePosition: 'left',
    };
    const wrapper = mount(<PromoHeadline content customFields={mockCustomFields} />);
    expect(wrapper).toMatchSnapshot();
  });
});
