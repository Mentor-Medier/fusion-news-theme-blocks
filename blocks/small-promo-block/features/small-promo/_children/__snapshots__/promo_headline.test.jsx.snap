// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promo headline return null while no content available in props 1`] = `
<PromoHeadline
  content={true}
  customFields={
    Object {
      "imagePosition": "left",
    }
  }
>
  <div
    className="promo-headline margin-md-left"
  >
    <a
      className="sm-promo-headline"
    >
      <styled.h2
        className="sm-promo-headline"
        contentEditable="true"
        suppressContentEditableWarning={true}
      >
        <StyledComponent
          className="sm-promo-headline"
          contentEditable="true"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "gDGmfb",
                "rules": Array [
                  "
  font-family: ",
                  [Function],
                  ";
",
                ],
              },
              "displayName": "styled.h2",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "h2",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          suppressContentEditableWarning={true}
        >
          <h2
            className="sc-bdVaJa gDGmfb sm-promo-headline"
            contentEditable="true"
            suppressContentEditableWarning={true}
          />
        </StyledComponent>
      </styled.h2>
    </a>
  </div>
</PromoHeadline>
`;
